modules/standard/Spawn.chpl

```diff
@@ -668,10 +748,53 @@ module Spawn {
   proc subprocess.communicate() {
     var err:syserr = ENOERR;
 
+    _halt_on_launch_error();
+
     this.communicate(error=err);
-    if err then halt("Error in subprocess.communicate: " + errorToString(err));
+    if err then ioerror(err, "in subprocess.communicate");
   }
 
+  /*
+    Close any open channels and pipes for interacting with a subprocess.  This
+    function does not wait for the subprocess to complete.  Note that it is
+    generally not necessary to call this function since these channels will be
+    closed when the subprocess record goes out of scope.
+
+    :arg error: optional argument to capture any error encountered
+                when closing the pipes.
+   */
+  proc subprocess.close(out error:syserr) {
+    error = ENOERR;
+    // Close stdin.
+    if this.stdin_pipe {
+      // send data to stdin
+      if this.stdin_buffering {
+        this.stdin._commit();
+        this.stdin_buffering = false; // Don't commit again on close again
+      }
```

Comment:
You do this exact if statement a couple of times, should there be a helper method?